src/modules/generateRandomNumber.js

```diff
@@ -7,7 +7,19 @@
  * @return {number} A random 4-digit number
  */
 function generateRandomNumber() {
-  /* Write your code here */
+  const digits = new Set();
+
+  const firstDigit = Math.floor(1 + Math.random() * 9);
+
+  digits.add(firstDigit);
+
+  while (digits.size < 4) {
+    const randomDigit = Math.floor(Math.random() * 10);
+
+    digits.add(randomDigit);
+  }
+
+  return Number(Array.from(digits).join(''));
```

Comment:
The current implementation converts the `Set` of digits directly to a number. This approach does not guarantee that the resulting number will always be a valid 4-digit number, as the order of digits in a `Set` is not controlled. Consider using an array to manage the order of digits explicitly, ensuring the first digit remains at the start.